### PR TITLE
Fix precision_recall_curve to correctly handle masked examples

### PR DESCRIPTION
**Pull request by dlin28**
The current precision_recall_curve computation is incorrect for weight=0 or masked examples.
The fix is to ignoring masked examples when computing tp, fp, and thresholds in binary_clf_curve.
Added more tests for checking masked examples are handled correctly.

In addition, fix precision_recall_curve to correctly handle weights and added tests as well.

### CODE REVIEW

File: axlearn/common/metrics_classification.py [hunk 1]
```diff
@@ -172,6 +177,9 @@ def binary_clf_curve(
     fps = jnp.where(y_pred_diff, fps, jnp.iinfo(jnp.int32).max)
     fps = jax.lax.cummin(fps, reverse=True)
 
+    # Masked entries at the end of thresholds are set to jnp.finfo(jnp.float32).max.
+    # Reset them to the rightmost unmasked threshold value.
+    thresholds = jnp.where(weight, thresholds, thresholds[jnp.argsort(weight)[-1]])
```
Comment (markblee) [inline axlearn/common/metrics_classification.py:182]: Thanks, as discussed offline maybe we should remove this to simplify, since the masked values should not be used anyway, and I'm not sure that `jnp.argsort(weight)[-1]` gives the rightmost unmasked value.

Comment (dlin28) [reply axlearn/common/metrics_classification.py]: Done

**Review by markblee — APPROVED** [review]
